src/strategyHelpers/blueGreen/deploy.ts

```diff
@@ -0,0 +1,139 @@
+import * as core from '@actions/core'
+
+import {Kubectl} from '../../types/kubectl'
+import {RouteStrategy} from '../../types/routeStrategy'
+
+import {
+   BlueGreenDeployment,
+   BlueGreenManifests,
+   deployWithLabel,
+   getManifestObjects,
+   GREEN_LABEL_VALUE,
+   deployObjects
+} from './blueGreenHelper'
+import {setupSMI} from './smiBlueGreenHelper'
+
+import {routeBlueGreenForDeploy} from './route'
+
+export async function deployBlueGreen(
+   kubectl: Kubectl,
+   files: string[],
+   routeStrategy: RouteStrategy
+): Promise<BlueGreenDeployment> {
+   let blueGreenDeployment: BlueGreenDeployment
+
+   if (routeStrategy == RouteStrategy.INGRESS) {
+      blueGreenDeployment = await Promise.resolve(
+         deployBlueGreenIngress(kubectl, files)
+      )
+   } else if (routeStrategy == RouteStrategy.SMI) {
+      blueGreenDeployment = await Promise.resolve(
+         deployBlueGreenSMI(kubectl, files)
+      )
+   } else {
+      blueGreenDeployment = await Promise.resolve(
+         deployBlueGreenService(kubectl, files)
+      )
+   }
+
+   core.startGroup('Routing blue green')
+   await routeBlueGreenForDeploy(kubectl, files, routeStrategy)
+   core.endGroup()
+   core.debug(
+      'objects deployed for ' +
+         routeStrategy +
+         ': ' +
+         JSON.stringify(blueGreenDeployment.objects)
```

Comment:
prefer template strings to concatenating with +

``hello ${world}`` is easier to read and faster. Plus you avoid weird javascript bugs with +